synapse/config/server.py

```diff
@@ -302,6 +302,24 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
                 "public_baseurl cannot contain query parameters or a #-fragment"
             )
 
+        self.extra_well_known_content = config.get("extra_well_known_content")
```

Comment:
We need a fallback here (I think this was missed from an earlier comment). Without it, if `extra_well_known_content` is not specified then `self.extra_well_known_content is None` and the type(...) check will fail. I'm assuming this explains the test failures.

```suggestion
        self.extra_well_known_content = config.get("extra_well_known_content", {})
```